plugins/opusCodec/src/OpusEncoder.cpp

```diff
@@ -0,0 +1,269 @@
+
+#include <PerfStat.h>
+#include <QtCore/QLoggingCategory>
+
+#include "OpusEncoder.h"
+#include "opus/opus.h"
+
+static QLoggingCategory encoder("AthenaOpusEncoder");
+
+static QString error_to_string(int error) {
+    switch (error) {
+    case OPUS_OK:
+        return "OK";
+    case OPUS_BAD_ARG:
+        return "One or more invalid/out of range arguments.";
+    case OPUS_BUFFER_TOO_SMALL:
+        return "The mode struct passed is invalid.";
+    case OPUS_INTERNAL_ERROR:
+        return "An internal error was detected.";
+    case OPUS_INVALID_PACKET:
+        return "The compressed data passed is corrupted.";
+    case OPUS_UNIMPLEMENTED:
+        return "Invalid/unsupported request number.";
+    case OPUS_INVALID_STATE:
+        return "An encoder or decoder structure is invalid or already freed.";
+    default:
+        return QString("Unknown error code: %i").arg(error);
+    }
+}
+
+
+
+AthenaOpusEncoder::AthenaOpusEncoder(int sampleRate, int numChannels)
+{
+    _opus_sample_rate = sampleRate;
+    _opus_channels    = numChannels;
+
+    int error;
+
+    _encoder = opus_encoder_create(sampleRate, numChannels, DEFAULT_APPLICATION, &error);
+
+    if ( error != OPUS_OK ) {
+        qCCritical(encoder) << "Failed to initialize Opus encoder: " << error_to_string(error);
+        _encoder = nullptr;
+        return;
+    }
+
+    setBitrate(DEFAULT_BITRATE);
+    setComplexity(DEFAULT_COMPLEXITY);
+    setApplication(DEFAULT_APPLICATION);
+    setSignal(DEFAULT_SIGNAL);
+
+    qCDebug(encoder) << "Opus encoder initialized, sampleRate = " << sampleRate << "; numChannels = " << numChannels;
+}
+
+AthenaOpusEncoder::~AthenaOpusEncoder()
+{
```

Comment:
Please put { on same line as function name CS 4.1.4 https://github.com/kasenvr/project-athena/blob/kasen/core/CODING_STANDARD.md#414-method-definitions-should-have-the-following-form